iothub/service/src/Feedback/MessageFeedbackProcessorClient.cs

```diff
@@ -10,36 +10,61 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.Amqp;
-using Microsoft.Azure.Amqp.Encoding;
-using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Amqp;
 using Microsoft.Azure.Devices.Common;
 using Microsoft.Azure.Devices.Common.Exceptions;
-using Microsoft.Azure.Devices.Common.Extensions;
 
 namespace Microsoft.Azure.Devices
 {
     /// <summary>
     /// Subclient of <see cref="IotHubServiceClient"/> for receiving cloud-to-device message feedback.
     /// </summary>
     /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d"/>.
-    public class MessageFeedbackProcessorClient : IDisposable
+    public class MessageFeedbackProcessorClient
     {
         private readonly string _hostName;
         private readonly IotHubConnectionProperties _credentialProvider;
-        private readonly IotHubConnection _connection;
-        private readonly AmqpFeedbackReceiver _feedbackReceiver;
+        private readonly AmqpConnectionHandler _amqpConnection;
 
         /// <summary>
         /// The callback to be executed each time message feedback is received from the service.
         /// </summary>
         /// <remarks>
         /// May not be null.
         /// </remarks>
+        /// <example>
+        /// serviceClient.MessageFeedbackProcessor.MessageFeedbackProcessor = OnFeedbackReceived;
+        /// serviceClient.MessageFeedbackProcessor.OpenAsync();
+        ///
+        /// //...
+        ///
+        /// public AcknowledgementType OnFeedbackReceived(FeedbackBatch feedbackBatch)
+        /// {
+        ///    foreach (FeedbackRecord record in feedback.Records)
+        ///    {
+        ///        Console.WriteLine($"Received feedback from device {record.DeviceId}")
+        ///    }
+        ///
+        ///    return AcknowledgementType.Complete;
+        /// }
+        /// </example>
         public Func<FeedbackBatch, AcknowledgementType> MessageFeedbackProcessor;
```

Comment:
We're using a property instead of a setter method (like elsewhere)?

Reply:
I'll make this a setter, sure